perception/autoware_tensorrt_bevdet/src/cpu_jpegdecoder.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024 AutoCore, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "autoware/tensorrt_bevdet/cpu_jpegdecoder.hpp"
+
+#include "autoware/tensorrt_bevdet/preprocess.hpp"
+
+#include <cuda_runtime.h>
+#include <jpeglib.h>
+#include <setjmp.h>
+#include <stddef.h>
+#include <stdio.h>
+
+#include <vector>
+
+struct jpeg_error_handler
+{
+  struct jpeg_error_mgr pub;
+  jmp_buf setjmp_buffer;
+};
+
+int decode_jpeg(const std::vector<char> & buffer, uchar * output)
+{
+  struct jpeg_decompress_struct cinfo;
+  struct jpeg_error_handler jerr;
+  cinfo.err = jpeg_std_error(&jerr.pub);
+
+  jerr.pub.error_exit = [](j_common_ptr cinfo) {
+    jpeg_error_handler * myerr = reinterpret_cast<jpeg_error_handler *>(cinfo->err);
+    longjmp(myerr->setjmp_buffer, 1);
+  };
+
+  if (setjmp(jerr.setjmp_buffer)) {
+    RCLCPP_ERROR(
+      rclcpp::get_logger("decode_jpeg"), "Failed to decompress jpeg: %s",
```

Comment:
Consider shortening this line for better readability. You could potentially split the RCLCPP_ERROR macro into multiple lines.